app.js

```diff
@@ -121,9 +123,20 @@ app.post('/api/registerCurrentData', async (req, res) => {
 })
 
 app.get('/api/getData', async (req, res) => {
-  const fileToSend = await getHighestPriorityFile()
+  let fileToSend
+  // the query data determines which priority we have: newest or oldest first.
+  const priority = req.query.data
+  if (priority === 'old') {
+    console.log('1')
+    fileToSend = await getHighestPriorityFile(true)
+  } else if (priority === 'new') {
+    console.log('2')
+    fileToSend = await getHighestPriorityFile(false)
+  } else {
+    console.log('3')
+    fileToSend = -1
+  }
   if (fileToSend === -1) {
```

Comment:
Anstatt es auf -1 zu setzten, könne man auch auf null checken

Reply:
Eigt ist ein check hier insgesamt unnötig und du könntest den Body der if Abfrage ab 139 in die den else Body ab 135 packen. Dann sparst du dir den check of -1 oder null.